runtime/vam/agg.go

```diff
@@ -77,29 +99,50 @@ func (c *countByString) countFixed(vec *vector.Const) {
 	if zed.TypeUnder(val.Type()) == zed.TypeString {
 		c.table[zed.DecodeString(val.Bytes())] += uint64(vec.Length())
 	}
+	if zed.TypeUnder(val.Type()) == zed.TypeNull {
+		c.nulls += uint64(vec.Length())
+	}
 }
 
 func (c *countByString) materialize(zctx *zed.Context, name string) *vector.Record {
 	typ := zctx.MustLookupTypeRecord([]zed.Field{
 		{Type: zed.TypeString, Name: name},
 		{Type: zed.TypeUint64, Name: "count"},
 	})
-	counts := make([]uint64, len(c.table))
-	keys := make([]string, len(c.table))
-	var off int
+	length := len(c.table)
+	counts := make([]uint64, length)
+	var bytes []byte
+	offs := make([]uint32, length+1)
+	var k int
 	for key, count := range c.table {
-		keys[off] = key
-		counts[off] = count
-		off++
+		offs[k] = uint32(len(bytes))
+		bytes = append(bytes, []byte(key)...)
+		counts[k] = count
+		k++
+	}
+	offs[k] = uint32(len(bytes))
+	// XXX change nulls to string null... this will be fixed in
+	// prod-quality summarize op
+	var nulls *vector.Bool
+	if c.nulls > 0 {
+		length++
+		counts = slices.Grow(counts, length)[0:length]
+		offs = slices.Grow(offs, length+1)[0 : length+1]
+		counts[k] = c.nulls
+		k++
+		offs[k] = uint32(len(bytes))
+		nulls = vector.NewBoolEmpty(uint32(k), nil)
+		nulls.Set(uint32(k - 1))
```

Comment:
This code is all going away and I futzed with it to bit to get it to work.  It's just for the hacky demo and there's no test coverage so I'd rather not manually not have to check that the change works for just right now.  It will disappear soon (vam-summarize branch).